src/test/resources/com/puppycrawl/tools/checkstyle/grammar/java14/InputJava14TextBlocks.txt

```diff
@@ -0,0 +1,277 @@
+    |       |   |--TYPE -> TYPE [8:8]
+    |       |   |   `--IDENT -> String [8:8]
+    |       |   |--IDENT -> doubleQuotes [8:15]
+    |       |   `--ASSIGN -> = [8:28]
+    |       |       `--EXPR -> EXPR [8:30]
+    |       |           `--TEXT_BLOCK_LITERAL_BEGIN -> """ [8:30]
+    |       |               |--TEXT_BLOCK_CONTENT -> \n                ""\n                 [8:33]
+    |       |               `--TEXT_BLOCK_LITERAL_END -> """ [10:16]
+    |       |--SEMI -> ; [10:19]
+    |       |--VARIABLE_DEF -> VARIABLE_DEF [11:8]
+    |       |   |--MODIFIERS -> MODIFIERS [11:8]
+    |       |   |--TYPE -> TYPE [11:8]
+    |       |   |   `--IDENT -> String [11:8]
+    |       |   |--IDENT -> oneDoubleQuote [11:15]
+    |       |   `--ASSIGN -> = [11:30]
+    |       |       `--EXPR -> EXPR [11:32]
+    |       |           `--TEXT_BLOCK_LITERAL_BEGIN -> """ [11:32]
+    |       |               |--TEXT_BLOCK_CONTENT -> \n                "\n                "\n                 [11:35]
+    |       |               `--TEXT_BLOCK_LITERAL_END -> """ [14:16]
+    |       |--SEMI -> ; [14:19]
+    |       |--VARIABLE_DEF -> VARIABLE_DEF [15:8]
+    |       |   |--MODIFIERS -> MODIFIERS [15:8]
+    |       |   |--TYPE -> TYPE [15:8]
+    |       |   |   `--IDENT -> String [15:8]
+    |       |   |--IDENT -> empty [15:15]
+    |       |   `--ASSIGN -> = [15:21]
+    |       |       `--EXPR -> EXPR [15:23]
+    |       |           `--TEXT_BLOCK_LITERAL_BEGIN -> """ [15:23]
+    |       |               |--TEXT_BLOCK_CONTENT -> \n                 [15:26]
+    |       |               `--TEXT_BLOCK_LITERAL_END -> """ [16:16]
+    |       |--SEMI -> ; [16:19]
+    |       |--VARIABLE_DEF -> VARIABLE_DEF [17:8]
+    |       |   |--MODIFIERS -> MODIFIERS [17:8]
+    |       |   |--TYPE -> TYPE [17:8]
+    |       |   |   `--IDENT -> String [17:8]
+    |       |   |--IDENT -> oneSingleQuote [17:15]
+    |       |   `--ASSIGN -> = [17:30]
+    |       |       `--EXPR -> EXPR [17:32]
+    |       |           `--TEXT_BLOCK_LITERAL_BEGIN -> """ [17:32]
+    |       |               |--TEXT_BLOCK_CONTENT -> \n                '\n                 [17:35]
+    |       |               `--TEXT_BLOCK_LITERAL_END -> """ [19:16]
+    |       |--SEMI -> ; [19:19]
+    |       |--VARIABLE_DEF -> VARIABLE_DEF [20:8]
+    |       |   |--MODIFIERS -> MODIFIERS [20:8]
+    |       |   |--TYPE -> TYPE [20:8]
+    |       |   |   `--IDENT -> String [20:8]
+    |       |   |--IDENT -> escape [20:15]
+    |       |   `--ASSIGN -> = [20:22]
+    |       |       `--EXPR -> EXPR [20:24]
+    |       |           `--TEXT_BLOCK_LITERAL_BEGIN -> """ [20:24]
+    |       |               |--TEXT_BLOCK_CONTENT -> \n                <html>\u000D\u000A\n\n                    <body>\u000D\u000A\n\n                        <p>Hello, world</p>\u000D\u000A\n\n                    </body>\u000D\u000A\n\n                </html>\u000D\u000A\n                 [20:27]
```

Comment:
Why aren't these double escaped? If a newline is printed as `\n`, it seems to me `\` should be printed as `\\`.
This seems like an issue to me.

Reply:
I'm not sure that I follow.

>If a newline is printed as `\n`

Both the text block characters and escape sequences are being printed exactly how the content is written in the java file, except for one exception; the actual line terminators ( `\n` in this case) are being added to the "end" of each line. 

>it seems to me` \` should be printed as `\\`

I don't understand why we should add an additional `\`, can you explain?

Maybe this will aid in our discussion:
```
➜  text-blocks-grammar /usr/lib/jvm/java-14-openjdk/bin/javac --enable-preview --source 14 TestClass.java
Note: TestClass.java uses preview language features.
Note: Recompile with -Xlint:preview for details.
➜  text-blocks-grammar cat TestClass.java
public class TestClass {
	String escape1 = "<html>\\u000D\\u000A\\n" +
		      "           <body>\\u000D\\u000A\\n" +
		      "               <p>Hello, world</p>\\u000D\\u000A\\n" +
		      "           </body>\\u000D\\u000A\\n" +
		      "       </html>\\u000D\\u000A";

	String escape2 = """
                      <html>\u000D\u000A\n
                          <body>\u000D\u000A\n
                              <p>Hello, world</p>\u000D\u000A\n
                          </body>\u000D\u000A\n
                      </html>\u000D\u000A
        """;
}
➜  text-blocks-grammar java $RUN_LOCALE -jar ~/IdeaProjects/checkstyle/target/checkstyle-8.36-SNAPSHOT-all.jar -t TestClass.java
CLASS_DEF -> CLASS_DEF [1:0]
|--MODIFIERS -> MODIFIERS [1:0]
|   `--LITERAL_PUBLIC -> public [1:0]
|--LITERAL_CLASS -> class [1:7]
|--IDENT -> TestClass [1:13]
`--OBJBLOCK -> OBJBLOCK [1:23]
    |--LCURLY -> { [1:23]
    |--VARIABLE_DEF -> VARIABLE_DEF [2:1]
    |   |--MODIFIERS -> MODIFIERS [2:1]
    |   |--TYPE -> TYPE [2:1]
    |   |   `--IDENT -> String [2:1]
    |   |--IDENT -> escape1 [2:8]
    |   |--ASSIGN -> = [2:16]
    |   |   `--EXPR -> EXPR [5:46]
    |   |       `--PLUS -> + [5:46]
    |   |           |--PLUS -> + [4:62]
    |   |           |   |--PLUS -> + [3:45]
    |   |           |   |   |--PLUS -> + [2:44]
    |   |           |   |   |   |--STRING_LITERAL -> "<html>\\u000D\\u000A\\n" [2:18]
    |   |           |   |   |   `--STRING_LITERAL -> "           <body>\\u000D\\u000A\\n" [3:8]
    |   |           |   |   `--STRING_LITERAL -> "               <p>Hello, world</p>\\u000D\\u000A\\n" [4:8]
    |   |           |   `--STRING_LITERAL -> "           </body>\\u000D\\u000A\\n" [5:8]
    |   |           `--STRING_LITERAL -> "       </html>\\u000D\\u000A" [6:8]
    |   `--SEMI -> ; [6:38]
    |--VARIABLE_DEF -> VARIABLE_DEF [8:1]
    |   |--MODIFIERS -> MODIFIERS [8:1]
    |   |--TYPE -> TYPE [8:1]
    |   |   `--IDENT -> String [8:1]
    |   |--IDENT -> escape2 [8:8]
    |   |--ASSIGN -> = [8:16]
    |   |   `--EXPR -> EXPR [8:18]
    |   |       `--TEXT_BLOCK_LITERAL_BEGIN -> """ [8:18]
    |   |           |--TEXT_BLOCK_CONTENT -> \n                      <html>\u000D\u000A\n\n                          <body>\u000D\u000A\n\n                              <p>Hello, world</p>\u000D\u000A\n\n                          </body>\u000D\u000A\n\n                      </html>\u000D\u000A\n         [8:21]
    |   |           `--TEXT_BLOCK_LITERAL_END -> """ [14:8]
    |   `--SEMI -> ; [14:11]
    `--RCURLY -> } [15:0]

```

Reply:
Sorry for the confusion, my point was content in the AST.

We escaped special characters. `\n`, `\r`, etc.. I would expect everything to follow this escaping even single `\` as it will make parsing by others easier.

If someone put in `String s = """\\""";` I would expect the AST to to say `TEXT_BLOCK_CONTENT -> \\`.

This would make it universally clear that all `\`s are escape characters for the first instance seen.

Imagine if someone put in `String s = """\\n""";`. As you have your AST generating now, it would say `TEXT_BLOCK_CONTENT -> \n`. How would we parse this then? By your other rules this would mean the string is purely a new line, and wouldn't be parsed as actual text characters `\` and `n`.

We should escape all `\` text seen in the contents.

Reply:
>Imagine if someone put in `String s = """\\n""";`

Note, this is not allowed.  You must have a newline after the opening delimiter.   I'm not sure if that matters in this context, but I just wanted to point that out.   Here is some more output to consider:
```
➜  text-blocks-grammar /usr/lib/jvm/java-14-openjdk/bin/javac --enable-preview --source 14 TestClass.java
Note: TestClass.java uses preview language features.
Note: Recompile with -Xlint:preview for details.

➜  text-blocks-grammar cat TestClass.java 
public class TestClass {

	String s1 = """
            \\n""";
	String s2 = """
            \n""";
	String s3 = """
            \\""";
	String s4 = "\\n";
	String s5 = "\n";
	String s6 = "\\";
}
                                                                                                                           
 ➜  text-blocks-grammar java $RUN_LOCALE -jar ~/IdeaProjects/checkstyle/target/checkstyle-8.36-SNAPSHOT-all.jar -t TestClass.java
CLASS_DEF -> CLASS_DEF [1:0]
|--MODIFIERS -> MODIFIERS [1:0]
|   `--LITERAL_PUBLIC -> public [1:0]
|--LITERAL_CLASS -> class [1:7]
|--IDENT -> TestClass [1:13]
`--OBJBLOCK -> OBJBLOCK [1:23]
    |--LCURLY -> { [1:23]
    |--VARIABLE_DEF -> VARIABLE_DEF [3:1]
    |   |--MODIFIERS -> MODIFIERS [3:1]
    |   |--TYPE -> TYPE [3:1]
    |   |   `--IDENT -> String [3:1]
    |   |--IDENT -> s1 [3:8]
    |   |--ASSIGN -> = [3:11]
    |   |   `--EXPR -> EXPR [3:13]
    |   |       `--TEXT_BLOCK_LITERAL_BEGIN -> """ [3:13]
    |   |           |--TEXT_BLOCK_CONTENT -> \n            \\n [3:16]
    |   |           `--TEXT_BLOCK_LITERAL_END -> """ [4:15]
    |   `--SEMI -> ; [4:18]
    |--VARIABLE_DEF -> VARIABLE_DEF [5:1]
    |   |--MODIFIERS -> MODIFIERS [5:1]
    |   |--TYPE -> TYPE [5:1]
    |   |   `--IDENT -> String [5:1]
    |   |--IDENT -> s2 [5:8]
    |   |--ASSIGN -> = [5:11]
    |   |   `--EXPR -> EXPR [5:13]
    |   |       `--TEXT_BLOCK_LITERAL_BEGIN -> """ [5:13]
    |   |           |--TEXT_BLOCK_CONTENT -> \n            \n [5:16]
    |   |           `--TEXT_BLOCK_LITERAL_END -> """ [6:14]
    |   `--SEMI -> ; [6:17]
    |--VARIABLE_DEF -> VARIABLE_DEF [7:1]
    |   |--MODIFIERS -> MODIFIERS [7:1]
    |   |--TYPE -> TYPE [7:1]
    |   |   `--IDENT -> String [7:1]
    |   |--IDENT -> s3 [7:8]
    |   |--ASSIGN -> = [7:11]
    |   |   `--EXPR -> EXPR [7:13]
    |   |       `--TEXT_BLOCK_LITERAL_BEGIN -> """ [7:13]
    |   |           |--TEXT_BLOCK_CONTENT -> \n            \\ [7:16]
    |   |           `--TEXT_BLOCK_LITERAL_END -> """ [8:14]
    |   `--SEMI -> ; [8:17]
    |--VARIABLE_DEF -> VARIABLE_DEF [9:1]
    |   |--MODIFIERS -> MODIFIERS [9:1]
    |   |--TYPE -> TYPE [9:1]
    |   |   `--IDENT -> String [9:1]
    |   |--IDENT -> s4 [9:8]
    |   |--ASSIGN -> = [9:11]
    |   |   `--EXPR -> EXPR [9:13]
    |   |       `--STRING_LITERAL -> "\\n" [9:13]
    |   `--SEMI -> ; [9:18]
    |--VARIABLE_DEF -> VARIABLE_DEF [10:1]
    |   |--MODIFIERS -> MODIFIERS [10:1]
    |   |--TYPE -> TYPE [10:1]
    |   |   `--IDENT -> String [10:1]
    |   |--IDENT -> s5 [10:8]
    |   |--ASSIGN -> = [10:11]
    |   |   `--EXPR -> EXPR [10:13]
    |   |       `--STRING_LITERAL -> "\n" [10:13]
    |   `--SEMI -> ; [10:17]
    |--VARIABLE_DEF -> VARIABLE_DEF [11:1]
    |   |--MODIFIERS -> MODIFIERS [11:1]
    |   |--TYPE -> TYPE [11:1]
    |   |   `--IDENT -> String [11:1]
    |   |--IDENT -> s6 [11:8]
    |   |--ASSIGN -> = [11:11]
    |   |   `--EXPR -> EXPR [11:13]
    |   |       `--STRING_LITERAL -> "\\" [11:13]
    |   `--SEMI -> ; [11:17]
    `--RCURLY -> } [12:0]

```

Reply:
The above strings, printed:
```
/home/nick/.jdks/openjdk-14.0.1/bin/java --enable-preview -javaagent:/home/nick/.local/share/JetBrains/Toolbox/apps/IDEA-C/ch-0/201.8538.31/lib/idea_rt.jar=46221:/home/nick/.local/share/JetBrains/Toolbox/apps/IDEA-C/ch-0/201.8538.31/bin -Dfile.encoding=UTF-8 -classpath /home/nick/Desktop/full-record-grammar/out/production/check-updates TestClass
TextBlocks:
\n


\
String literals:
\n


\

```

Reply:
parser should not escape anything, all should be as is in file. Yes, we have hard choice with invisible "\n" symbol that we should make visible in result string.
 
@nmancus1 , please create simple use cases in input file
```
String s = """
                     \u000D"""
String s1 = """
                     \n"""
String s2 = """
                     \n
                  """
``` 

and lets look on AST,
no matter what we will decide, this are good inputs to stay forever.

Reply:
> it seems to me \ should be printed as \\

please add also this to inputs.
We should not escape anything (only invisible '\n' is excpetion)

Reply:
Inputs are updated.